### PR TITLE
Pre commit: add poetry hooks and use them in mypy

### PR DESCRIPTION
**Pull request by ilaner**
## Related Issues
fixes:

## Description
* add update-additional-dependencies hook, which will automatically update the `.pre-commit-config.yaml` file according to the poetry.lock.
* Add additional dependencies to `mypy`.
* Add poetry built in hooks to validate our files.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,41 +1,216 @@
...
-          - id: mypy
+- repo: https://github.com/charliermarsh/ruff-pre-commit
+  rev: "v0.0.262"
+  hooks:
+  - id: ruff
+    args: [--fix, --exit-non-zero-on-fix]
+- repo: https://github.com/python-poetry/poetry
+  rev: "1.5.1"
+  hooks:
+  - id: poetry-check
```
Comment (dorschw) [inline .pre-commit-config.yaml:11]: Do we need both `check` and `lock --check`?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,41 +1,216 @@
...
+  - id: poetry-lock
+    args: [--check]
+  - id: poetry-export
+    args: [--without-hashes, "-o", "requirements.txt"]
```
Comment (dorschw) [inline .pre-commit-config.yaml:15]: Please add `requirements.txt` to `.gitignore` so nobody commits it by mistake

File: demisto_sdk/scripts/update_additional_dependencies.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+from demisto_sdk.commands.common.logger import logger
+from demisto_sdk.commands.common.tools import get_file
+
+yaml = YAML_Handler()
+
+
+def update_additional_dependencies(
+    pre_commit_config_path: Path, requirements_path: Path, hooks: Sequence[str]
+) -> int:
+    try:
```
Comment (dorschw) [inline demisto_sdk/scripts/update_additional_dependencies.py:16]: please document

File: demisto_sdk/scripts/update_additional_dependencies.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        pre_commit = get_file(pre_commit_config_path)
+        for repo in pre_commit["repos"]:
+            for hook in repo["hooks"]:
+                if hook["id"] in hooks:
+                    hook["additional_dependencies"] = requirements
+        with pre_commit_config_path.open("w") as f:
+            yaml.dump(pre_commit, f)
+        return 0
+    except Exception as e:
+        logger.error("Failed to update additional dependencies: %s", e)
```
Comment (dorschw) [inline demisto_sdk/scripts/update_additional_dependencies.py:32]:
```suggestion
        logger.exception("Failed to update additional dependencies")
```

File: demisto_sdk/scripts/update_additional_dependencies.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return 1
```
Comment (dorschw) [inline demisto_sdk/scripts/update_additional_dependencies.py:33]: why return 0/1 and `SystemExit` with it? isn't raising preferred?

File: demisto_sdk/scripts/update_additional_dependencies.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def main(argv: Optional[Sequence[str]] = None) -> int:
+    parser = argparse.ArgumentParser(
+        description="Update the additional dependencies of precommit"
```
Comment (dorschw) [inline demisto_sdk/scripts/update_additional_dependencies.py:38]:
```suggestion
        description="Update the additional dependencies of precommit hooks"
```

File: demisto_sdk/scripts/update_additional_dependencies.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    )
+    parser.add_argument(
+        "--pre_commit_config",
+        help="The path to the pre-commit config file",
```
Comment (dorschw) [inline demisto_sdk/scripts/update_additional_dependencies.py:42]:
```suggestion
        help="Path to the pre-commit config file",
```

File: demisto_sdk/scripts/update_additional_dependencies.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        default=".pre-commit-config.yaml",
+    )
+    parser.add_argument(
+        "--requirements",
+        help="The path to the requirements file",
+        default="requirements.txt",
+    )
+    parser.add_argument(
+        "hooks", help="The hooks to update", nargs="*", default=["mypy"]
```
Comment (dorschw) [inline demisto_sdk/scripts/update_additional_dependencies.py:51]: Are these comma-separated? or in what format?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (ilaner) [reply .pre-commit-config.yaml:15]: it is already in gitignore

File: demisto_sdk/scripts/update_additional_dependencies.py (hunk 5 above)
Comment (ilaner) [reply demisto_sdk/scripts/update_additional_dependencies.py]: inspired by the hooks here: https://github.com/pre-commit/pre-commit-hooks/tree/main/pre_commit_hooks

File: demisto_sdk/scripts/update_additional_dependencies.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        help="The path to the pre-commit config file",
+        default=".pre-commit-config.yaml",
+    )
+    parser.add_argument(
+        "--requirements",
+        help="Path to the requirements file",
+        default="requirements.txt",
+    )
+    parser.add_argument(
+        "hooks", help="The hooks to update", nargs="*", default=["mypy"]
```
Comment (dorschw) [inline demisto_sdk/scripts/update_additional_dependencies.py:61]:
```suggestion
        "hooks", help="The hooks to update, separated by spaces", nargs="*", default=["mypy"]
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ilaner) [reply .pre-commit-config.yaml:11]: Yes
poetry check: The check command validates the structure of the pyproject.toml file and returns a detailed report if there are any errors.

poetry lock --check: Verify that poetry.lock is consistent with pyproject.toml

**Review by dorschw — APPROVED** [review]

**Pull request merged**